common/src/main/java/io/pravega/common/tracing/RPCTracingHelpers.java

```diff
@@ -0,0 +1,106 @@
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ForwardingClientCallListener;
+import io.grpc.ForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class contains utilities to intercept RPC calls the server and client sides, as well as to track incoming
+ * requests for intra-component tracing.
+ */
+@Slf4j
+public final class RPCTracingHelpers {
+
+    private static final String REQUEST_DESCRIPTOR = "requestDescriptor";
+    private static final String REQUEST_ID = "requestId";
+    public static final CallOptions.Key<String> REQUEST_DESCRIPTOR_CALL_OPTION = CallOptions.Key.of(REQUEST_DESCRIPTOR, "");
+    public static final CallOptions.Key<String> REQUEST_ID_CALL_OPTION = CallOptions.Key.of(REQUEST_ID, "");
+    private static final Metadata.Key<String> REQUEST_DESCRIPTOR_HEADER = Metadata.Key.of(REQUEST_DESCRIPTOR, Metadata.ASCII_STRING_MARSHALLER);
+    private static final Metadata.Key<String> REQUEST_ID_HEADER = Metadata.Key.of(REQUEST_ID, Metadata.ASCII_STRING_MARSHALLER);
+
+    public static ClientInterceptor getClientInterceptor() {
+        return new FishTaggingClientInterceptor();
+    }
+
+    public static ServerInterceptor getServerInterceptor() {
+        return new FishTaggingServerInterceptor();
+    }
+
+    /**
+     * This interceptor is intended to get request tags from call options and attach them to the RPC request.
+     */
+    private static class FishTaggingClientInterceptor implements ClientInterceptor {
+
+        @Override
+        public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+
+            return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+                @Override
+                public void start(Listener<RespT> responseListener, Metadata headers) {
+                    // Get call options previously set by the client and attach that information to the RPC request.
+                    final String requestDescriptor = callOptions.getOption(REQUEST_DESCRIPTOR_CALL_OPTION);
+                    final String requestId = callOptions.getOption(REQUEST_ID_CALL_OPTION);
+
+                    if (requestDescriptor != null && requestId != null && !requestDescriptor.isEmpty() && !requestId.isEmpty()) {
+                        headers.put(REQUEST_DESCRIPTOR_HEADER, requestDescriptor);
+                        headers.put(REQUEST_ID_HEADER, requestId);
+                        log.info("[requestId={}] Tagging RPC request {}.", requestId, requestDescriptor);
+                    } else {
+                        log.warn("Not tagging request {}: Call options not containing request tags.", method.getFullMethodName());
```

Comment:
Suggested rephrase: `Not tagging request` -> `Request not tagged`.

Also, why is it at warn level?